### PR TITLE
Rewrite site backend in Rust

### LINKED ISSUE #176: Rewrite the backend in Rust
This is going to be a fairly large project that will probably take a while to do correctly. The ultimate goal of this project is to rewrite the code for this website in Rust using [warp](https://github.com/seanmonstar/warp). I will also fully nixify the CI/CD pipeline of this site. No feature will be left behind. No links will be broken. Everything will look and feel 1:1 the same as the original site, including compatibility with `lynx`.

These crates will be useful:

- https://crates.io/crates/sitemap
- https://crates.io/crates/robots_txt
- https://docs.rs/extract-frontmatter/1.0.2/extract_frontmatter/
- https://crates.io/crates/rss
- https://crates.io/crates/jsonapi
- https://github.com/seanmonstar/warp
- https://docs.rs/prometheus/0.9.0/prometheus/
- https://crates.io/crates/serde_dhall
- https://docs.rs/handlebars/3.2.1/handlebars/
- https://crates.io/crates/comrak

I will need to fork these crates into the repo:

- [x] https://github.com/pwoolcoc/jsonfeed

I will need to create these crates:

- [ ] patreon API client (might not do this?)
- [x] Go vanity import redirector

I will also need to do the following:

- [x] nixify static content, gallery, blog and talks folders into their own packages (cache efficiency)
- [x] adapt CD pipeline for Nix
- [x] Run CI for the whole program
- [x] Investigate the best way to inject secrets into kubermemes manifests (`env:FOO as Text`?)
- [ ] Adapt [kubermemes](https://tulpa.dev/cadey/kubermemes) for Kubernetes job generation
- [x] Adapt the site to take a dhall file as configuration

Site features:

- [x] `X-Clacks-Overhead`
- [x] `X-Hacker`
- [x] RSS/Atom/JSONFeed support
- [x] Front matter parsing (yaml, ick)
- [x] Blogpost rendering
- [x] Gallery rendering
- [x] Talk rendering
- [x] Signal boost page
- [ ] Patron page
- [x] Resume page
- [x] Contact page
- [x] Feeds page
- [x] Links to other sites (maybe have a links page?)
- [x] Prometheus metrics
- [ ] Sitemap
- [x] Adapt all templates to ructe

Basically see [here](https://github.com/Xe/site/blob/master/cmd/site/main.go#L252-L291)

Extra fun:

- [ ] Make tags actually searchable
- [x] Maintain the `christine.website/jsonfeed` package elsewhere and make a go vanity import redirector point to that so I don't break `mi`
- [ ] Embed link to active git commit in the footer of each page
- [ ] Automatically enable snow.css during winter
- [ ] Dial out to GitHub and [tulpa.dev](https://tulpa.dev), then make newsfeed entries for new releases of my projects.

### PR DESCRIPTION
**Pull request by Xe**
Closes #176 

This is gonna be a big one.

I will need to fork these crates into the repo:

- [x] https://github.com/pwoolcoc/jsonfeed

I will need to create these crates:

- [x] patreon API client (might not do this?)
- [x] Go vanity import redirector

I will also need to do the following:

- [x] nixify static content, gallery, blog and talks folders into their own packages (cache efficiency)
- [x] adapt CD pipeline for Nix
- [x] Run CI for the whole program
- [x] Investigate the best way to inject secrets into kubermemes manifests (`env:FOO as Text`?)
- [x] Adapt the site to take a dhall file as configuration
- [x] Delete the Go implementation

Site features:

- [x] `X-Clacks-Overhead`
- [x] `X-Hacker`
- [x] Atom support
- [x] RSS support
- [x] Jsonfeed support
- [x] Front matter parsing (yaml, ick)
- [x] Blog index
- [x] Blogpost rendering
- [x] Gallery index
- [x] Gallery rendering
- [x] Talk index
- [x] Talk rendering
- [x] Index page
- [x] Signal boost page
- [x] Patron page
- [x] Resume page
- [x] Contact page
- [x] Feeds page
- [x] Links to other sites (maybe have a links page?)
- [x] Prometheus metrics
- [x] Sitemap
- [x] Adapt all templates to ructe
- [x] Date each post

Basically see [here](https://github.com/Xe/site/blob/master/cmd/site/main.go#L252-L291)

Extra fun:

- [x] Maintain the `christine.website/jsonfeed` package elsewhere and make a go vanity import redirector point to that so I don't break `mi`
- [x] Embed link to active git commit in the footer of each page
- [x] Automatically enable snow.css during winter

### CODE REVIEW

**Comment by Xe** [comment]
https://devel.christine.website/
